feat: update and align the naming and argments in `from/to_hub/disk` methods

davidberenstein1957 opened the pull request:
# Description
<!-- Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

Closes #5286

- Rename `target_workspace` and `target_name` to `workspace` and `name`.
- Add `with_records` to all aligned methods.
- Update the order of `name` and `workspace` within `from/to` methods  to align with the default `rg.Dataset` 
- Update documentation https://github.com/argilla-io/argilla/pull/5285

**Type of change**
<!--  Please delete options that are not relevant. Remember to title the PR according to the type of change  -->

- Refactor (change restructuring the codebase without changing functionality)
- Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**
<!--  Please add some reference about how your feature has been tested.  -->

**Checklist**
<!--  Please go over the list and make sure you've taken everything into account -->

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

davidberenstein1957 committed:
Update `with_records` usage
Rename `target_name` and `target_workspace` to `name` and `workspace`

davidberenstein1957 committed:
Update phrasing and layout for documentation import export (#5285)

# Description
<!-- Please include a summary of the changes and the related issue.
Please also include relevant motivation and context. List any
dependencies that are required for this change. -->

Closes #5284 

NEW DOCS:
<img width="1325" alt="image"
src="https://github.com/user-attachments/assets/174f831d-5b4a-433d-9198-dfeccd93586f">

OLD DOCS:
<img width="1385" alt="image"
src="https://github.com/user-attachments/assets/14198174-b48d-45b7-bf1d-b3596b2f4258">

**Type of change**
<!-- Please delete options that are not relevant. Remember to title the
PR according to the type of change -->

- Documentation update

**How Has This Been Tested**
<!-- Please add some reference about how your feature has been tested.
-->

**Checklist**
<!-- Please go over the list and make sure you've taken everything into
account -->

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature
works
- I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

---------

Co-authored-by: Sara Han <127759186+sdiazlor@users.noreply.github.com>

frascuchon commented on argilla/src/argilla/datasets/_export/_hub.py:
```diff
@@ -61,8 +62,9 @@ def to_hub(
 
-        hfds = self.records(with_vectors=True, with_responses=True, with_suggestions=True).to_datasets()
-        hfds.push_to_hub(repo_id, **kwargs)
+        if with_records:
```
Can we create an HF dataset without records? Is this supported?

frascuchon replied:
I miss some integration tests checking to_hub / from_hub without records.

davidberenstein1957 replied:
@frascuchon thanks for the prompt to create tests. The original code also failed due to uploading files to `.argilla/*.json`. I currently aligned this for both local and Hugging Face Hub file structure.

davidberenstein1957 replied:
Also, as mentioned, created this. https://huggingface.co/collections/argilla-internal-testing/tests-integration-669fd0f70181df726ed508be

frascuchon commented on argilla/src/argilla/datasets/_export/_hub.py:
```diff
@@ -38,6 +37,7 @@ def to_hub(
         self: "Dataset",
         repo_id: str,
         *,
+        with_records: bool = True,
```
`to_hub` method only accepts the `repo_id` as positional argument. Maybe we should align this with `from_hub` method.

frascuchon commented on argilla/src/argilla/datasets/_export/_hub.py:
```diff
@@ -98,7 +100,8 @@ def to_hub(
         cls: Type["Dataset"],
         repo_id: str,
-        workspace: Optional[Union["Workspace", str, UUID]] = None,
+        name: Optional[str] = None,
+        workspace: Optional[Union["Workspace", str]] = None,
```
?
```suggestion
        *,
        name: Optional[str] = None,
        workspace: Optional[Union["Workspace", str]] = None,
```

davidberenstein1957 replied:
@frascuchon it was an intended change to update the order of the arguments so that each of them are shown in the same way.

davidberenstein1957 committed:
Apply suggestions from code review

Co-authored-by: Paco Aranda <frascuchon@gmail.com>

davidberenstein1957 committed:
Update argilla/src/argilla/datasets/_export/_hub.py

Co-authored-by: Paco Aranda <frascuchon@gmail.com>

davidberenstein1957 committed:
Update pdm script command for server deployments
Add integration tests
Add HF_TOKEN_ARGILLA_INTERNAL_TESTING as env var to tests
Update to align `.argilla` path for settings in local and remote mode
Fix error getting if from `dataset._client.me`

davidberenstein1957 committed:
Update leading naming `dev-server` to `server-dev` to align with leading naming

frascuchon commented on .github/workflows/argilla.yml:
```diff
@@ -53,6 +53,10 @@ jobs:
           while ! curl -XGET http://localhost:6900/api/_status; do sleep 5; done
+      - name: Set huggingface hub credentials
+        run: |
+          echo "HF_TOKEN_ARGILLA_INTERNAL_TESTING=${{ secrets.HF_TOKEN_ARGILLA_INTERNAL_TESTING }}" >> "$GITHUB_ENV"
```
IF the [`HF_TOKEN`](https://huggingface.co/docs/huggingface_hub/en/package_reference/environment_variables#hftoken) is the standard environment variable name, what if we create it based on the CI secrets?
```suggestion
          echo "HF_TOKEN=${{ secrets.HF_TOKEN_ARGILLA_INTERNAL_TESTING }}" >> "$GITHUB_ENV"
```

frascuchon commented on argilla/tests/integration/test_export_dataset.py:
```diff
@@ -63,63 +64,97 @@ def test_export_dataset_to_disk(dataset: rg.Dataset):
-    assert new_dataset.settings.questions[0].name == "label"
+@pytest.fixture
+def token():
+    return os.getenv("HF_TOKEN_ARGILLA_INTERNAL_TESTING") or os.getenv("HF_TOKEN")
```
Do we need this if we define the `HF_TOKEN` environment variable ?

davidberenstein1957 replied:
You are right it might be better to just include one but I feel weird about using `HF_TOKEN` since people will normally not have access to this and our developers would need to potentially work with different tokens which might be annoying.

frascuchon replied:
Maybe we can use the [pdm environment file](https://pdm-project.org/latest/usage/scripts/#env_file) for that. We already do for running tests with an specific argilla db https://github.com/argilla-io/argilla/blob/develop/argilla-server/pyproject.toml#L172

frascuchon replied:
Or in case we want avoid this ambiguity we can use only the `HF_TOKEN_ARGILLA_INTERNAL_TESTING` or `HF_TOKEN_TESTING`?

davidberenstein1957 replied:
I will just use HF_TOKEN_ARGILLA_INTERNAL_TESTING

frascuchon commented on argilla/tests/integration/test_export_dataset.py:
```diff
@@ -63,63 +64,97 @@ def test_export_dataset_to_disk(dataset: rg.Dataset):
+
+
+@pytest.mark.skipif(
+    not (os.getenv("HF_TOKEN_ARGILLA_INTERNAL_TESTING") or os.getenv("HF_TOKEN")),
```
Maybe is simpler using the standard naming.
```suggestion
    not os.getenv("HF_TOKEN"),
```

frascuchon commented on argilla/tests/integration/test_export_dataset.py:
```diff
@@ -63,63 +64,97 @@ def test_export_dataset_to_disk(dataset: rg.Dataset):
+    ):
+        repo_id = f"argilla-internal-testing/test_export_dataset_to_hub_with_records_{with_records}"
+        dataset.records.log(records=mock_data)
+        dataset.to_hub(repo_id=repo_id, with_records=with_records, token=token)
```
If the expected behavior is to read private repositories if the `HF_TOKEN` env var is defined, it makes sense to define a test where the token is skiped and to_hub/from_hub read internally the `HF_TOKEN` (I can imagine this is a `datasets` package feature.

davidberenstein1957 replied:
Yes, normally it reads automatically based on an internal token. I don't think it makes sense to add this additionally because this is a functionality from the Hugging Face hub/datasets.

frascuchon commented on argilla/tests/integration/test_export_dataset.py:
```diff
@@ -63,63 +64,97 @@ def test_export_dataset_to_disk(dataset: rg.Dataset):
+            assert exported_settings["fields"][0]["name"] == "text"
+            assert exported_settings["questions"][0]["name"] == "label"
+
+            assert exported_dataset["name"] == dataset.name
```
Should we add some checks when `with_records = False`?

frascuchon commented on argilla/tests/integration/test_export_dataset.py:
```diff
@@ -63,63 +64,97 @@ def test_export_dataset_to_disk(dataset: rg.Dataset):
+    ):
+        repo_id = f"argilla-internal-testing/test_export_dataset_to_hub_with_records_{with_records}"
+        dataset.records.log(records=mock_data)
+        dataset.to_hub(repo_id=repo_id, with_records=with_records, token=token)
```
Should we add some checks on this to validate the expected data in the repo? Or maybe an integration test with from_hub(to_hub)?

davidberenstein1957 replied:
I think this is captured in the `test_import_dataset_from_hub`

frascuchon commented on argilla/tests/integration/test_export_dataset.py:
```diff
@@ -63,63 +64,97 @@ def test_export_dataset_to_disk(dataset: rg.Dataset):
+    def test_import_dataset_from_hub(
+        self, token: str, dataset: rg.Dataset, client, mock_data: List[dict[str, Any]], with_records: bool
+    ):
+        repo_id = f"argilla-internal-testing/test_import_dataset_from_hub_with_records_{with_records}"
```
Are we testing a scenario where the remote repo has records and `with_records = False` and vice-versa?

davidberenstein1957 replied:
I will.

frascuchon commented on argilla-server/pyproject.toml:
```diff
@@ -167,8 +164,13 @@ line-length = 120
-server = { cmd = "uvicorn argilla_server:app --reload" }
+server = { cmd = "uvicorn argilla_server:app --port 6900 --reload" }
 migrate = { cmd = "alembic upgrade head" }
+server-dev.composite = [
+    "migrate",
+    "cli database users create_default",
+    "server"
+]
```
Cool! thanks for this.

Maybe it makes sense to update the [README](https://github.com/argilla-io/argilla/tree/feat/5286-feature-align-the-naming-and-accepted-params-fromto_diskhub-methods/argilla-server#run-uvicorn-development-server) with this ?

davidberenstein1957 committed:
Update reade with pdm dev server info

davidberenstein1957 committed:
Update tests to be incldue both true and false record export and import
Update solely relying on `HF_TOKEN_ARGILLA_INTERNAL_TESTING`

davidberenstein1957 committed:
Update skip if to `HF_TOKEN_ARGILLA_INTERNAL_TESTING`

davidberenstein1957 committed:
Fix tests TestDiskImportExportMixin::test_export_dataset_to_disk

davidberenstein1957 committed:
Update to include client during `rg.Dataset.from_hub`

frascuchon commented on argilla-server/README.md:
````diff
@@ -70,24 +70,17 @@ Folders inside `/api` are organized by API version, having right now only v1 imp
 
 ```sh
-pdm migrate
+server-dev
````
Do we not need to add the `pdm run`?
```suggestion
pdm run server-dev
```

davidberenstein1957 committed:
Update argilla-server/README.md

Co-authored-by: Paco Aranda <frascuchon@gmail.com>

frascuchon commented on argilla/src/argilla/datasets/_export/_hub.py:
```diff
@@ -46,13 +45,14 @@ def to_hub(
+            with_records: whether to load the records from the Hugging Face dataset. Defaults to `True`.
             generate_card: whether to generate a dataset card for the `Dataset` in the Hugging Face Hub. Defaults
                 to `True`.
             *args: the args to pass to `datasets.Dataset.push_to_hub`.
```
There is no `args` argument
```suggestion
```

davidberenstein1957 committed:
Update README.md

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

davidberenstein1957 committed:
Update _hub.py

frascuchon commented on argilla/src/argilla/datasets/_export/_hub.py:
```diff
@@ -107,7 +118,8 @@ def from_hub(
+            workspace (Union[Workspace, str], optional): The workspace to import the dataset to. Defaults to None and default workspace is used.
             client: the client to use to load the `Dataset`. If not provided, the default client will be used.
             with_records: whether to load the records from the Hugging Face dataset. Defaults to `True`.
             *args: the args to pass to `datasets.Dataset.load_from_hub`.
```
```suggestion
```

frascuchon commented on argilla/src/argilla/datasets/_export/_hub.py:
```diff
@@ -107,38 +117,48 @@ def from_hub(
             repo_id=repo_id,
             repo_type="dataset",
-            allow_patterns=cls._DEFAULT_CONFIGURATION_FILES,
+            allow_patterns=cls._default_configuration_files,
```
I think constants should be defined in UPPERCASE

frascuchon commented on argilla/src/argilla/datasets/_export/_hub.py:
```diff
@@ -61,20 +60,29 @@ def to_hub(
+            hfds = self.records(with_vectors=True, with_responses=True, with_suggestions=True).to_datasets()
+            hfds.push_to_hub(repo_id, **kwargs)
+        else:
+            hf_api.create_repo(repo_id=repo_id, repo_type="dataset", exist_ok=kwargs.get("exist_ok") or True)
```
Just a comment here. Since this will create an empty dataset repo. Not sure how often would be this scenario and if makes sense for the HF platform. Maybe this would be something to discuss with the Hub team.

davidberenstein1957 replied:
I am not sure how common this is either but given that we offer the `with_records` within the `to_disk`, I think it is better to align and include it or remove it from both the `to_disk` and `to_hub`.

frascuchon commented on argilla/src/argilla/datasets/_export/_hub.py:
```diff
@@ -162,7 +182,7 @@ def _log_dataset_records(hf_dataset: "HFDataset", dataset: "Dataset"):
         known_users_ids = [user.id for user in dataset._client.users]
         unknown_user_ids = set(user_ids.keys()) - set(known_users_ids)
-        my_user = dataset._client.me.id
+        my_user = dataset._client.me
```
This is a but, right? In that case, we should have a test to verify is already solved.

davidberenstein1957 replied:
This is captured in the integration tests because initially there were no tests included.

davidberenstein1957 committed:
Update constants to UPPER_CASE

frascuchon commented on argilla/src/argilla/datasets/_export/_disk.py:
```diff
@@ -12,38 +12,40 @@
-    _default_settings_path = "settings.json"
-    _default_records_path = "records.json"
-    _default_dataset_path = "dataset.json"
+    _MODEL: DatasetModel
```
I think this is not a constant since no value has been assigned.

```suggestion
    _model: DatasetModel
```

frascuchon commented on argilla/src/argilla/datasets/_export/_disk.py:
```diff
@@ -117,7 +123,7 @@ def _persist_dataset_model(self, path: Path):
             raise FileExistsError(f"Dataset already exists at {path}")
         with open(file=path, mode="w") as f:
-            json.dump(self._model.model_dump(), f)
+            json.dump(self._MODEL.model_dump(), f)
```
This won't work. The _model attribute is defined at "Dataset" level. Also, it would be better to user the `.api_model()`
```suggestion
            json.dump(self.api_model().model_dump(), f)
```

davidberenstein1957 committed:
Fix failing tests wit unused arguments

davidberenstein1957 committed:
Update argilla/src/argilla/datasets/_export/_disk.py

Co-authored-by: Paco Aranda <frascuchon@gmail.com>
